tests/test_cli_compile.py

```diff
@@ -858,6 +858,27 @@ def test_upgrade_packages_option_no_existing_file(pip_conf, runner):
     assert out.exit_code == 0
     assert "small-fake-a==0.2" in out.stderr.splitlines()
     assert "small-fake-b==0.3" in out.stderr.splitlines()
+    assert (
+        "WARNING: the output file requirements.txt exists but is empty"
+        not in out.stderr
+    )
+
+
+def test_upgrade_packages_empty_target_file_warning(pip_conf, runner):
```

Comment:
It seems we're missing tests for corner cases like `/dev/null` and `-`/stdout here...

Could you verify these for me, please? It may be useful to stick `@pytest.mark.parametrize` here.

Reply:
If the user does a shell redirect (`pip-compile ... > /dev/null`), there's no impact, because `pip-compile`'s normal logic for finding its output file applies.

If the user tries to do `pip-compile -o /dev/null`, they receive an error before the code-path that's changed in this PR, as that path can't be opened for writing.

I'm not sure I understand your suggestion regarding stdout. Could you say more?

Reply:
I wasn't sure either, that's why I asked. I'll mark this thread as resolved.